File: ui/commands/integer_test.js
// Copyright (C) 2019-2026 Ni Rui <ranqus@gmail.com>
// Copyright (C) 2026 Snuffy2
// SPDX-License-Identifier: AGPL-3.0-only

import assert from "assert";
import * as reader from "../stream/reader.js";
import * as integer from "./integer.js";

describe("Integer", () => {
  it("Integer 127", async () => {
    let i = new integer.Integer(127),
      marshalled = i.marshal();

    let r = new reader.Reader(new reader.Multiple(() => {}), (data) => {
      return data;
    });

    assert.strictEqual(marshalled.length, 1);

    r.feed(marshalled);

    let i2 = new integer.Integer(0);

    await i2.unmarshal(r);

    assert.strictEqual(i.value(), i2.value());
  });

  it("Integer MAX", async () => {
    let i = new integer.Integer(integer.MAX),
      marshalled = i.marshal();

    let r = new reader.Reader(new reader.Multiple(() => {}), (data) => {
      return data;
    });

    assert.strictEqual(marshalled.length, 2);

    r.feed(marshalled);

    let i2 = new integer.Integer(0);

    await i2.unmarshal(r);

    assert.strictEqual(i.value(), i2.value());
  });
});

File: application/commands/integer_test.go
// Copyright (C) 2019-2026 Ni Rui <ranqus@gmail.com>
// Copyright (C) 2026 Snuffy2
// SPDX-License-Identifier: AGPL-3.0-only

package commands

import (
	"bytes"
	"testing"
)

func TestInteger(t *testing.T) {
	ii := Integer(0x3fff)
	result := Integer(0)
	buf := make([]byte, 2)

	mLen, mErr := ii.Marshal(buf)

	if mErr != nil {
		t.Error("Failed to marshal:", mErr)

		return
	}

	mData := bytes.NewBuffer(buf[:mLen])
	mErr = result.Unmarshal(mData.Read)

	if mErr != nil {
		t.Error("Failed to unmarshal:", mErr)

		return
	}

	if result != ii {
		t.Errorf("Expecting result to be %d, got %d instead", ii, result)

		return
	}
}

func TestIntegerSingleByte1(t *testing.T) {
	ii := Integer(102)
	result := Integer(0)
	buf := make([]byte, 2)

	mLen, mErr := ii.Marshal(buf)

	if mErr != nil {
		t.Error("Failed to marshal:", mErr)

		return
	}

	if mLen != 1 {
		t.Errorf("Expecting the Integer to be marshalled into %d bytes, got "+
			"%d instead", 1, mLen)

		return
	}

	mData := bytes.NewBuffer(buf[:mLen])
	mErr = result.Unmarshal(mData.Read)

	if mErr != nil {
		t.Error("Failed to unmarshal:", mErr)

		return
	}

	if result != ii {
		t.Errorf("Expecting result to be %d, got %d instead", ii, result)

		return
	}
}

func TestIntegerSingleByte2(t *testing.T) {
	ii := Integer(127)
	result := Integer(0)
	buf := make([]byte, 2)

	mLen, mErr := ii.Marshal(buf)

	if mErr != nil {
		t.Error("Failed to marshal:", mErr)

		return
	}

	if mLen != 1 {
		t.Errorf("Expecting the Integer to be marshalled into %d bytes, got "+
			"%d instead", 1, mLen)

		return
	}

	mData := bytes.NewBuffer(buf[:mLen])
	mErr = result.Unmarshal(mData.Read)

	if mErr != nil {
		t.Error("Failed to unmarshal:", mErr)

		return
	}

	if result != ii {
		t.Errorf("Expecting result to be %d, got %d instead", ii, result)

		return
	}
}
